libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2252,3 +2252,140 @@ string YulUtilFunctions::revertReasonIfDebug(string const& _message)
 {
 	return revertReasonIfDebug(m_revertStrings, _message);
 }
+
+string YulUtilFunctions::decodeReturnParametersFunction(
+	TypePointers const& _returnTypes,
+	bool _dynamicReturnSize
+)
+{
+	ABIFunctions abi(m_evmVersion, m_revertStrings, m_functionCollector);
+
+	string const abiDecode = abi.tupleDecoder(_returnTypes, true);
+	string const functionName =
+		"decode_return_parameters_" +
+		revertStringsToString(m_revertStrings) + "_" +
+		abiDecode;
+
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return util::Whiskers(R"(
+			function <functionName>(dataMpos, returnSize) <?hasRetVars> -> <retVars> </hasRetVars> {
```

Comment:
Due to the peculiar handling of the dataMpos parameter here and since this function is now only used once, we might consider inlining it again.

Reply:
Yeah... inlining would be fine, if used only once - or we should eliminate the ``dataMpos`` argument and just read the free memory pointer again inside the function.

Reply:
(eliminate it in the dynamic case only that is... if that works... maybe inlining is actually better)

Reply:
"just read the free memory pointer again" - it is still a weird implicit assumption.